### ISSUE TITLE
Unclear meaning of "debug session"

### ISSUE THREAD

**Issue by mickaelistria** [Microsoft/debug-adapter-protocol#386]
I'm trying to implement support for `startDebugging` in Eclipse LSP4E, using the dapDebugServer.js build by vscode-js-debug.
But after many attempts, I didn't yet manage to figure out from the spec and the code what should be done to properly handle the `startDebugging` request.
The specification mentions "debug session", but what is a "debug session" exactly? Particularly, how do I start a new "debug session"? Should I send initialized again, or send only launch; should I reuse the same connection or spawn a new debug adapter...? It would be great if the spec could clarify when does a new debug session start, and how to start one as a reaction of `startDebugging`.

**Comment by puremourning** [comment]
> Should I send initialized again, or send only launch

this is easy. You should perform the full sequence as if a completely independent debug adapter, i.e. you should send initilialze,get response, sent launch (or attach depending on the startDebugging request), and the usual breakpoints, initializationDone, etc. dance.

So each "session" is completely independent.

**Comment by mickaelistria** [comment]
> this is easy to answer. You should perform the full sequence as if a completely independent debug adapter

OK thanks. It was my first interpretation from the spec; so it looks like the spec is clear enough on what to send; what is less clear is where or how to send it; and it looks like this remain adapter-specific.
The issue you link are closed. Would it make sense to open a new one to make the spec clearer on how both server and clients should handle multi-sessions instead and fill the gap so there is no more room for interpretation?

**Comment by puremourning** [comment]
I agree that further clarification is required in the spec, as the only 2 implementations I've seen so far don't seem to agree on its interpretation, and at least 3 independent client authors (including you and me) struggled to get them working.
